File: LeetCode/Simulation/Golang/Leetcode-slidingwindowmean.go
package sbgo

type MovingAverage struct {
	sum, size int
	arr       []int
}

/** Initialize your data structure here. */
func Constructor(size int) MovingAverage {
	sum, arr, size := 0, []int{}, size
	return MovingAverage{sum, size, arr}
}

func (this *MovingAverage) Next(val int) float64 {
	this.arr = append(this.arr, val)
	this.sum += val
	if len(this.arr) > this.size {
		this.sum -= this.arr[0]
		this.arr = this.arr[1:]
	}
	return float64(this.sum) / float64(len(this.arr))
}

File: LeetCode/Simulation/Rust/Leetcode-slidingwindowmean.rs
use std::collections::VecDeque;

struct MovingAverage {
    sum:i32,
    arr:VecDeque<i32>,
    size:usize
}


/**
 * `&self` means the method takes an immutable reference.
 * If you need a mutable reference, change it to `&mut self` instead.
 */
impl MovingAverage {

    /** Initialize your data structure here. */
    fn new(size: i32) -> Self {
        MovingAverage{sum : 0,arr : VecDeque::new(),size: size as usize}
    }
    
    fn next(&mut self, val: i32) -> f64 {
        self.arr.push_back(val);
        self.sum += val;
        if self.size < self.arr.len() {
            self.sum -= self.arr.pop_front().unwrap();
        }
        self.sum as f64 / self.arr.len() as f64
    }
}